test/pipelinefinally_test.go

```diff
@@ -266,10 +266,6 @@ spec:
 				t.Errorf("Error waiting for TaskRun to succeed: %v", err)
 			}
 			dagTask2EndTime = taskrunItem.Status.CompletionTime
-		case n == "dagtask3":
-			if !isSkipped(t, n, taskrunItem.Status.Conditions) {
-				t.Fatalf("dag task %s should have skipped due to condition failure", n)
-			}
```

Comment:
for reviewers' convenience, "dagtask3" was migrated to use when expressions and is in skipped tasks section of pipelinerun status now:

https://github.com/tektoncd/pipeline/pull/5027/files#diff-7815ee756d1326fe39c09f4deb5c7c36aa4a8f14b4319d5bf8ac73a1929a3d10L342-L349